core/src/main/kotlin/org/eln2/sim/thermal/main.kt

```diff
@@ -0,0 +1,146 @@
+import kotlinx.coroutines.*
+import kotlinx.coroutines.sync.*
+import kotlin.math.*
+
+class ThermalInterface{
+
+    var mutex = Mutex()
+    var R = 1f //heat resistance at the interface
+    var heatTranfer = 0f //heat transfer during a simulation tick through this interface
+
+    var lastTickComputed = 0 //number of the last tick where the heat transfer was computed. Tick number switch between 0 and 1. This is here to prevent computing twice the heat trasnfered when not needed
+
+
+    fun computeHeatTransfer(dt : Float, tickNumber : Int){
+        var T0 = e0.getT(tickNumber)
+        var T1 = e1.getT(tickNumber)
+        if(tickNumber != lastTickComputed) {
+            heatTranfer = (T1 - T0) / R * dt
+            lastTickComputed = tickNumber
+        }
+    }
+
+    lateinit var e0 : ThermalElement
+    lateinit var e1 : ThermalElement
+}
+
+class SignedThermalInterface(thInterface : ThermalInterface, sgn : Float){
+    var mThermalInterface :ThermalInterface = thInterface
+    var sign : Float = sgn
+
+    fun computeHeatTransfer(dt : Float, tickNumber : Int):Float{
+        mThermalInterface.computeHeatTransfer(dt,tickNumber)
+        return sign*mThermalInterface.heatTranfer;
+    }
+}
+
+class ThermalElement{
+    var C = 1f; //Heat capacity of a given element
+    var P = 0f; //Power dissipated as heat in the thermal element (from electrical circuit)
```

Comment:
You shouldn't need the semicolons here; the Kotlin style guide seems to advise against them unless necessary

Reply:
C++ habits? :p

Reply:
C, C++, Rust... I do the same thing when I'm switching between languages too :P